src/Css/Stylesheet.php

```diff
@@ -769,6 +769,10 @@ private function _css_selector_to_xpath(string $selector, bool $first_pass = fal
                         $attr .= $tok[$j++];
                     }
 
+                    if (!isset($tok[$j])){
+                        break;
```

Comment:
Note that breaking here changes the logic, the existing `$query` remains; so e.g. the invalid selector `.w[href` would be treated as `.w` now instead of `.w[href]` as before. To just remove the warning, retaining the current logic, one could use `$tok[$j] = null;` instead.

It would be better to drop the entire selector instead. I am working on some small updates here to do just that.

Reply:
See #3057